### PR TITLE
Factor out Zc4PostingWriter from Zc4PostingSeqWrite.

### PR DESCRIPTION
**Pull request by toregge**
@geirst : please review

### CODE REVIEW

File: searchlib/src/vespa/searchlib/diskindex/zcposting.cpp [hunk 1]
```diff
@@ -813,83 +710,47 @@ Zc4PostingSeqWrite::open(const vespalib::string &name,
...
+    writeContext.setFile(&_file);
+    search::ComprBuffer &cb = writeContext;
+    EncodeContext &e = _writer.get_encode_context();
+    writeContext.allocComprBuf(65536u, 32768u);
+    e.setupWrite(cb);
+    // Reset accumulated stats
+    _fileBitSize = 0;
+    // Start write initial header
+    makeHeader(fileHeaderContext);
+//    _writer.get_encode_features().setupWrite(_writer.get_feature_write_context());
```
Comment (geirst) [inline searchlib/src/vespa/searchlib/diskindex/zcposting.cpp]: This line can be removed

File: searchlib/src/vespa/searchlib/test/fakedata/fakezcfilterocc.cpp [hunk 2]
```diff
@@ -181,288 +150,89 @@ FakeZcFilterOcc::setupT(const FakeWord &fw, bool doFeatures,
...
-        assert(static_cast<int32_t>(docIdDelta) > 0);
-        l3SkipBytes.encode(docIdDelta - 1);
-    }
-    if (l4SkipBytes.size() > 0) {
-        uint32_t docIdDelta = lastDocId - lastL4SkipDocId;
-        assert(static_cast<int32_t>(docIdDelta) > 0);
-        l4SkipBytes.encode(docIdDelta - 1);
     }
+    writer.flush_word();
+    // TODO:
```
Comment (geirst) [inline searchlib/src/vespa/searchlib/test/fakedata/fakezcfilterocc.cpp]: Remove this?

File: searchlib/src/vespa/searchlib/diskindex/zc4_posting_writer.cpp [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+
+    if (_encodeFeatures != nullptr) {
+        _encodeFeatures->flush();
+    }
+    EncodeContext &e = _encodeContext;
+
+    uint32_t numDocs = _docIds.size();
+
+    e.encodeExpGolomb(numDocs - 1, K_VALUE_ZCPOSTING_NUMDOCS);
+    if (numDocs >= _minChunkDocs)
```
Comment (geirst) [inline searchlib/src/vespa/searchlib/diskindex/zc4_posting_writer.cpp]: Consider use of "{"

File: searchlib/src/vespa/searchlib/diskindex/zc4_posting_writer_base.cpp [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+            }
+        }
+        uint32_t docId = dit->first;
+        featurePos += dit->second;
+        _zcDocIds.encode(docId - lastDocId - 1);
+        lastDocId = docId;
+        ++l1SkipCnt;
+    }
+    // Extra partial entries for skip tables to simplify iterator during search
+    if (_l1Skip.size() > 0)
```
Comment (geirst) [inline searchlib/src/vespa/searchlib/diskindex/zc4_posting_writer_base.cpp]: Consider using "{". Applies for next lines as well.

**Review by geirst — APPROVED** [review]
👍 Some nitpicks to consider

**Review by geirst — APPROVED** [review]
👍

**Comment by baldersheim** [comment]
Trigger merge stop enforcer

**Pull request merged**